sdk/core/Azure.Core/src/Serialization/ModelSerializer.cs

```diff
@@ -0,0 +1,203 @@
+        /// </summary>
+        /// <param name="model">The model to convert.</param>
+        /// <param name="format">The <see cref="ModelSerializerFormat"/> to use.</param>
+        /// <returns>A binary representation of the serialized model.</returns>
+        public static BinaryData ConvertToBinaryData(IModelJsonSerializable<object> model, ModelSerializerFormat format)
+            => ConvertToBinaryData(model, new ModelSerializerOptions(format));
+
+        /// <summary>
+        /// Converts an <see cref="IModelXmlSerializable{T}"/> into a <see cref="BinaryData"/>.
+        /// </summary>
+        /// <param name="model">The model to convert.</param>
+        /// <param name="options">The <see cref="ModelSerializerOptions"/> to use.</param>
+        /// <returns>A binary representation of the serialized model.</returns>
+        public static BinaryData ConvertToBinaryData(IModelXmlSerializable<object> model, ModelSerializerOptions? options = default)
+        {
+            options ??= ModelSerializerOptions.DefaultWireOptions;
+            using MemoryStream stream = new MemoryStream();
+            using XmlWriter writer = XmlWriter.Create(stream);
+            model.Serialize(writer, options);
+            writer.Flush();
+            return new BinaryData(stream.GetBuffer().AsMemory(0, (int)stream.Position));
+        }
+
+        /// <summary>
+        /// Converts an <see cref="IModelXmlSerializable{T}"/> into a <see cref="BinaryData"/>.
+        /// </summary>
+        /// <param name="model">The model to convert.</param>
+        /// <param name="format">The <see cref="ModelSerializerFormat"/> to use.</param>
+        /// <returns>A binary representation of the serialized model.</returns>
+        public static BinaryData ConvertToBinaryData(IModelXmlSerializable<object> model, ModelSerializerFormat format)
+            => ConvertToBinaryData(model, new ModelSerializerOptions(format));
+
+        private static IModelSerializable<object> GetInstance(Type returnType)
+        {
+            var model = GetObjectInstance(returnType) as IModelSerializable<object>;
+            if (model is null)
+                throw new InvalidOperationException($"{returnType.Name} does not implement {nameof(IModelSerializable<object>)}");
+            return model;
+        }
+
+        private static IModelSerializable<T> GetInstance<T>() where T : IModelSerializable<T>
+        {
+            var model = GetObjectInstance(typeof(T)) as IModelSerializable<T>;
+            if (model is null)
+                throw new InvalidOperationException($"{typeof(T).Name} does not implement {nameof(IModelSerializable<T>)}");
+            return model;
+        }
+
+        private static object GetObjectInstance(Type returnType)
+        {
+            var typeToActivate = returnType.IsAbstract ? returnType.Assembly.GetTypes().FirstOrDefault(t => t.Name == $"Unknown{returnType.Name}") : returnType;
```

Comment:
This is unreliable and very slow. Can't we resolve this via interface as well?

Reply:
do you mean add a method like this on the interface?
```c#
Type GetTypeToInstantiate();
```

Not sure if we want this but it is an interesting idea on how to avoid the convention Activator.

Reply:
I would probably prefer something along the lines of an attribute on the base type vs an API on the interface.

@tg-msft, @KrzysztofCwalina we discussed this portion of needing to drop to some kind of reflection in the abstract case quite a bit.  How do you guys feel about using an attribute here so finding the type is explicit and throwing if the attribute is not present?

Reply:
I don't think we can just add a method to the interface. We don't have an instance to call the method and static interface members are not NetStandard 2.0.

Reply:
We definitely shouldn't be enumerating all types.  That could end up breaking in cases like FR where we ship a second iteration of the library in the same package if we (accidentally?) reuse a name.  At a minimum, we should assume it's in the same namespace and use `Type.GetType($"{returnType}.Namespace}.Unknown{returnType.Name}")`;  We could also generate the `UnknownFoo` class as an internal nested class so it'd just be `returnType.GetNestedType("Unknown");` if we wanted to make things more performant.

I'd be fine switching to an internal attribute like `[UnknownImplementation(typeof(Chupacabra)] public abstract class Animal { ... }` too.  It'd be worth making sure we can combine this with something from https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis?view=net-7.0 to work with AOT though.